homeassistant/components/mqtt/mixins.py

```diff
@@ -404,8 +416,9 @@ def _async_setup_entities() -> None:
                 error = str(ex)
                 config_file = getattr(yaml_config, "__config_file__", "?")
                 line = getattr(yaml_config, "__line__", "?")
-                issue_id = hex(hash(frozenset(yaml_config.items())))
-                yaml_config_str = yaml.dump(dict(yaml_config))
+                issue_id = hex(hash(frozenset(yaml_config)))
+                # Remove additional info from dict before dumping to YAML
+                yaml_config_str = yaml.dump(json_loads(json_bytes(yaml_config)))
```

Comment:
What's the point of this?

Reply:
When yaml_config is formatted directly, then `yaml.dump` prints a lot of additional stuff about line numbers types. May be this is a bug in yaml.dump. To avoid this I serialize the config to a json first and then convert it back.

Reply:
Example print of the old code:
````
availability: !!python/object/new:homeassistant.util.yaml.objects.NodeListClass
  listitems:
  - !!python/object/new:homeassistant.util.yaml.objects.NodeDictClass
    dictitems:
      topic: test/av1
    state:
      __config_file__: /workspaces/core/config/configuration.yaml
      __line__: 138
  - !!python/object/new:homeassistant.util.yaml.objects.NodeDictClass
    dictitems:
      topic: test/av2
    state:
      __config_file__: /workspaces/core/config/configuration.yaml
      __line__: 139
  state:
    __config_file__: /workspaces/core/config/configuration.yaml
    __line__: 138
availability_mode: latest
bla: true
device: !!python/object/new:homeassistant.util.yaml.objects.NodeDictClass
  dictitems:
    identifiers: !!python/object/new:homeassistant.util.yaml.objects.NodeListClass
      listitems:
      - dev001
      state:
        __config_file__: /workspaces/core/config/configuration.yaml
        __line__: 135
    name: Bedroom temperature
  state:
    __config_file__: /workspaces/core/config/configuration.yaml
    __line__: 133
device_class: temperature
entity_category: config
state_topic: home/bedroom/temperature
unique_id: brtemp01
unit_of_measurement: "\xB0C"
```

Reply:
I see. Please explain this better in the comment, maybe:
# Remove unwanted additional info such as line numbers from config before dumping to YAML
Also, can you share an example of the unwanted output?

Reply:
Using the HA yaml dumper worked out of the box.